subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncher.java

```diff
@@ -201,6 +147,75 @@ public void stop() {
         }
     }
 
+    private class RunBuildAction implements Transformer<BuildResult, BuildOperationContext> {
+        private final Stage upTo;
+
+        private RunBuildAction(Stage upTo) {
+            this.upTo = upTo;
+        }
+
+        @Override
+        public BuildResult transform(BuildOperationContext buildOperationContext) {
+            BuildOperationWorkerRegistry.Completion workerLease = buildServices.get(BuildOperationWorkerRegistry.class).operationStart();
```

Comment:
I'm wondering in which case you _cannot_ use an action. This would imply that the returned completion "leaks", making it hard to track.